src/config/backend/EdFi.DmsConfigurationService.Backend.Postgresql/OpenIddict/PostgresTokenManager.cs

```diff
@@ -0,0 +1,691 @@
+            IEnumerable<KeyValuePair<string, string>> credentials)
+        {
+            try
+            {
+                string? clientId = null;
+                string? clientSecret = null;
+                string? scope = null;
+
+                foreach (var kvp in credentials)
+                {
+                    if (kvp.Key.Equals("client_id", StringComparison.OrdinalIgnoreCase))
+                    {
+                        clientId = kvp.Value;
+                    }
+                    else if (kvp.Key.Equals("client_secret", StringComparison.OrdinalIgnoreCase))
+                    {
+                        clientSecret = kvp.Value;
+                    }
+                    else if (kvp.Key.Equals("scope", StringComparison.OrdinalIgnoreCase))
+                    {
+                        scope = kvp.Value;
+                    }
+                }
+
+                if (string.IsNullOrWhiteSpace(clientId) || string.IsNullOrWhiteSpace(clientSecret))
+                {
+                    _logger.LogWarning("Missing client credentials in token request");
+                    return new TokenResult.FailureUnknown("Missing client_id or client_secret");
+                }
+
+                _logger.LogDebug(
+                    "Attempting to generate token for client: {ClientId}",
+                    SanitizeForLog(clientId)
+                );
+
+                await using var connection = new NpgsqlConnection(_databaseOptions.Value.DatabaseConnection);
+                await connection.OpenAsync();
+
+                // First, get the application by client ID to retrieve the stored (potentially hashed) secret
+                string applicationSql =
+                    @"SELECT a.Id, a.DisplayName, a.Permissions, a.ClientSecret,
+                             array_agg(s.Name) as Scopes
+                      FROM dmscs.OpenIddictApplication a
+                      LEFT JOIN dmscs.OpenIddictApplicationScope aps ON a.Id = aps.ApplicationId
+                      LEFT JOIN dmscs.OpenIddictScope s ON aps.ScopeId = s.Id
+                      WHERE a.ClientId = @ClientId
+                      GROUP BY a.Id, a.DisplayName, a.Permissions, a.ClientSecret";
+
+                var applicationInfo = await connection.QuerySingleOrDefaultAsync<ApplicationInfo>(
+                    applicationSql,
+                    new { ClientId = clientId }
+                );
+
+                if (applicationInfo == null)
+                {
+                    _logger.LogDebug(
+                        "Client not found: {ClientId}",
+                        SanitizeForLog(clientId)
+                    );
+                    return new TokenResult.FailureIdentityProvider(
+                            new IdentityProviderError.InvalidClient("Invalid client or Invalid client credentials"));
+                }
+                // Verify the client secret using the hasher (supports both hashed and plain text for backward compatibility)
```

Comment:
Why still support plaintext? If we really need to, we need a ticket to remove plaintext support later.

Reply:
I remove that, now is not supported